lib/puppet/util/package/version/pip.rb

```diff
@@ -0,0 +1,230 @@
+module Puppet::Util::Package::Version
+  class Pip
+    def initialize(version)
+      matched = version.match(Regexp.new(("^\\s*") + @@VERSION_PATTERN + ("\\s*$"), Regexp::EXTENDED | Regexp::MULTILINE | Regexp::IGNORECASE))
+
+      raise ValidationFailure.new(version) unless matched
+      self.class.include Comparable
```

Comment:
I thought it would be good to include Comparable only if the object will be created(string was parsed successfully) but I don't really know if there are any noticeable performance improvements.